src/coreclr/vm/jitinterface.cpp

```diff
@@ -11822,6 +11873,72 @@ void* CEEJitInfo::getFieldAddress(CORINFO_FIELD_HANDLE fieldHnd,
     return result;
 }
 
+bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t* buffer, int bufferSize)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    _ASSERT(fieldHnd != NULL);
+    _ASSERT(buffer != NULL);
+    _ASSERT(bufferSize > 0);
+
+    bool result = false;
+
+    JIT_TO_EE_TRANSITION();
+
+    FieldDesc* field = (FieldDesc*)fieldHnd;
+    _ASSERTE(field->IsStatic() && !field->IsThreadStatic());
+    _ASSERTE((unsigned)bufferSize == field->GetSize());
+
+    MethodTable* pEnclosingMT = field->GetEnclosingMethodTable();
+    if (!pEnclosingMT->IsSharedByGenericInstantiations())
```

Comment:
```suggestion
    _ASSERTE(!pEnclosingMT->IsSharedByGenericInstantiations());
    _ASSERTE(!pEnclosingMT->ContainsGenericVariables());
```
I think these can be asserts too. The JIT is not expected to call this for shared generics or types with generic variables - if it does, it means there is something seriously wrong.